### PR TITLE
refactor(ast): Uses block for CtIf and CtLoop.

### LINKED ISSUE #683: new feature: Statement always in a block
Always put a block in loop and condition body.

In order to handle loops and conditions without block, we can set the block as implicit and then not print it.

### PR DESCRIPTION
**Pull request by GerardPaligot**
Closes #683

### CODE REVIEW

File: src/test/java/spoon/test/intercession/IntercessionTest.java [hunk 1]
```diff
@@ -136,8 +136,8 @@ public void testInsertIfIntercession() {
 		CtIf ifStmt = (CtIf) foo.getBody().getStatements().get(0);
 		String s = ifStmt.toString().replace("\r", "");
 		assertEquals(ifCode, s);
-		CtReturn<?> r1 = (CtReturn<?>) ifStmt.getThenStatement();
-		CtReturn<?> r2 = (CtReturn<?>) ifStmt.getElseStatement();
+		CtBlock<?> r1 = ifStmt.getThenStatement();
```
Comment (monperrus) [inline src/test/java/spoon/test/intercession/IntercessionTest.java]: add assert on isImplicit?

**Comment by monperrus** [comment]
excellent!

could you add assert on toString to verify the pretty-printing of implicit blocks?

File: src/test/java/spoon/test/intercession/IntercessionTest.java (hunk 1 above)
Comment (GerardPaligot) [inline src/test/java/spoon/test/intercession/IntercessionTest.java]: Done.

**Comment by GerardPaligot** [comment]
> could you add assert on toString to verify the pretty-printing of implicit blocks?

There are so many asserts on that in our tests.

File: src/main/java/spoon/support/reflect/code/CtStatementImpl.java [hunk 2]
```diff
@@ -185,14 +185,16 @@ void insert(CtBlock<?> block, CtStatementList statementsToBeInserted) {
 			}
 
 			@Override
-			void insertFromFirstStatement(CtBlock<?> block, CtStatement target,
-					CtStatementList statementsToBeInserted) {
+			void insertFromFirstStatement(CtBlock<?> block, CtStatement target, CtStatementList statementsToBeInserted) {
```
Comment (GerardPaligot) [inline src/main/java/spoon/support/reflect/code/CtStatementImpl.java]: Done.

**Comment by GerardPaligot** [comment]
@monperrus PR ok

**Pull request merged**